capstone-rs/examples/recursive.rs

```diff
@@ -0,0 +1,87 @@
+use std::collections::{HashMap, VecDeque};
+use std::env;
+use std::fs;
+use std::process;
+
+use object::{Object, ObjectSection, SectionKind};
+
+use capstone;
+use capstone::arch::x86::X86Insn;
+use capstone::prelude::*;
+
+fn main() {
+    let args: Vec<_> = env::args().collect();
+    if args.len() != 2 {
+        eprintln!("Usage: {} <file>", args[0]);
+        process::exit(-1);
+    }
+
+    let buf = if let Ok(bd) = fs::read(&args[1]) {
+        bd
+    } else {
+        eprintln!("cannot read file");
+        process::exit(-1);
+    };
+
+    let obj = if let Ok(od) = object::File::parse(&*buf) {
+        od
+    } else {
+        eprintln!("cannot parse file");
+        process::exit(-1);
+    };
+
+    let mut addr_queue: VecDeque<u64> = VecDeque::new();
+    let mut addr_seen: HashMap<u64, bool> = HashMap::new();
+
+    for section in obj.sections() {
+        if section.kind() == SectionKind::Text {
+            println!("{:x?} ", section);
+            addr_queue.push_back(section.address());
+        }
+    }
+
+    let cs = if let Ok(cs) = Capstone::new()
+        .x86()
+        .mode(arch::x86::ArchMode::Mode64)
+        .detail(true)
+        .build()
+    {
+        cs
+    } else {
+        eprintln!("failed to create capstone handle");
+        process::exit(-1);
+    };
+
+    let mut disasm = cs.get_disasm_iter();
+
+    while !addr_queue.is_empty() {
```

Comment:
Instead of using `pop_front().unwrap()` inside the loop, you can use "while let" like:

```
while let Some(addr) = addr_queue.pop_front() {
    // ...
}
```

This is more efficient and avoids the need to `unwrap()`.